images/quayio-pull-through-cache/Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM quay.io/centos/centos:stream8
+RUN yum -y install httpd-tools wget jq && yum -y clean all  && rm -rf /var/cache
+
+
+RUN set -eux; \
+	not_parsed="$(curl -s https://api.github.com/repos/distribution/distribution/releases/latest | jq .name)"; \
+	version="${not_parsed:2:-1}"; \
+	arch="$(arch)"; \
+	case "$arch" in \
+		x86_64)  arch='amd64';; \
+		aarch64) arch='arm64';; \
+		armhf)   arch='armv6';; \
+		armv7)   arch='armv7';; \
+		ppc64le) arch='ppc64le';; \
+		s390x)   arch='s390x';; \
+		*) echo >&2 "error: unsupported architecture: $arch"; exit 1 ;; \
+	esac; \
+	wget -O registry.tar.gz "https://github.com/distribution/distribution/releases/download/v${version}/registry_${version}_linux_${arch}.tar.gz"; \
```

Comment:
We should exercise caution here. Major changes could take place in the implementation. Config options, server API, changes that require data migration / etc (e.g. when 3.0 comes up). Given the impacts of the service failing to start, I'd lean toward this being an intentional bump or at least constraining it to a major/minor.

Reply:
Docker hardcoded this to 2.8.1 and does a manual upgrade. We can do it similarly.

Reply:
Fixed to 2.8.1, updates are very rare anyway